src/lineup/ALineUpView2.ts

```diff
@@ -221,6 +235,112 @@ export abstract class ALineUpView2 extends AView {
     this.handleSelectionColumnsImpl(selection);
   }
 
+  private getColumnColor(id: number) : string {
+    let color = '';
+    if(!this.colorMap.has(id)) {
+      color = this.colors.shift();
+      this.colorMap.set(id, color);
+    } else {
+      color = this.colorMap.get(id);
+    }
+    return color;
+  }
+
+  private freeColumnColor(id: number) : void {
+    const color = this.colorMap.get(id);
+    if(color) {
+      this.colorMap.delete(id);
+      if(this.colors.indexOf(color) === -1) {
+        this.colors.unshift(color);
+      }
+    }
+  }
+
+  private addDynamicColumns(ids: number[]) : void {
+    const addColumn = (desc, newColumnPromise, id) => {
+      //mark as lazy loaded
+      (<any>desc).lazyLoaded = true;
+      this.withoutTracking(() => {
+        this.addColumn(desc, newColumnPromise, id, true); // true == withoutTracking
+      });
+    };
+
+    ids.forEach((id) => {
+      this.getSelectionColumnDesc(id)
+        .then((columnDesc) => {
+          // add multiple columns
+          if(!this.dynamicColumns.has(id)) {
+            this.dynamicColumns.set(id, new Set());
+          }
+          if(Array.isArray(columnDesc)) {
+            if(columnDesc.length > 0) {
+              // Save which columns have been added for a specific element in the selection
+              const selectedElements = new Set(columnDesc.map((desc) => desc.selectionOptions.id));
```

Comment:
the options field is not further specified (like in the utility functions but you are requesting that it is an object and has an id field) should be noted somewhere